Final/fdlist.h

```diff
@@ -0,0 +1,81 @@
+#ifndef FD_LIST_H
+#define FD_LIST_H
+
+#include <stdio.h>
+#include <stdlib.h>
+
+struct node {
```

Comment:
let's convert this into a linked list class as practice for creating classes. let's split this file into a `linked_list.h` and `linked_list.cpp`. here's some starter code of what the entire .h, and some of the .cpp:

linked_list.h
```
#ifndef LINKED_LIST_H
#define LINKED_LIST_H

typedef struct node {
    int data;
    struct node *next;
} node;

class LinkedList {
public:
    // Constructor
    LinkedList();
    // Destructor
    ~LinkedList();

    // Pushes node to front of list
    void push(int data);
    // Removes the first instance of a node with the given data
    void remove(int data);

    // Exposed to allow for easy iteration through the linked list:
    // for (node *cur = list.begin(); cur != list.end(); cur = cur->next) {
    //     // Do something
    // }
    node *begin();
    node *end();
private:
    node *head;
    node *sentinel;
};

#endif
```

linked_list.cpp
```
#include "linked_list.h"

// Create linked list
LinkedList::LinkedList()
{
    head = sentinel = (node *)calloc(1, sizeof(node));
}

// Clean up nodes
LinkedList::~LinkedList()
{
    node *cur = head;
    while (cur) {
        node *temp = cur->next;
        free(cur);
        cur = temp;
    }
}

void LinkedList::push(int data)
{
    // Fill this in
}

void LinkedList::remove(int data)
{
    // Fill this in
}

node * LinkedList::begin()
{
    return head;
}

node * LinkedList::end()
{
    return sentinel;
}
```